packages/typescript-estree/src/convert.ts

```diff
@@ -1145,6 +1145,13 @@ export class Converter {
         const isAbstract = hasModifier(SyntaxKind.AbstractKeyword, node);
         const isAccessor = hasModifier(SyntaxKind.AccessorKeyword, node);
 
+        if (isAbstract && node.initializer) {
+          this.#throwError(
+            node.initializer,
+            `Abstract property cannot have an initializer.`,
+          );
+        }
+
```

Comment:
Nit: no need to compute `isAccessor` if it's going to throw an error anyway:

```suggestion

        if (isAbstract && node.initializer) {
          this.#throwError(
            node.initializer,
            `Abstract property cannot have an initializer.`,
          );
        }

        const isAccessor = hasModifier(SyntaxKind.AccessorKeyword, node);
```